### PR TITLE
add client side filtering recipe

### PR DESCRIPTION
**Pull request by tomkralidis**
cc @antje-s @6a6d74 @golfvert, as discussed at ET-W2IT 2024-11-11.

### CODE REVIEW

**Review by antje-s — APPROVED** [review]
thanks for the updates, looks good only one double "additional" in line 155 to be deleted

File: cookbook/sections/data-publishers.adoc [hunk 1]
```diff
@@ -118,5 +118,187 @@ image::images/data-publishers-wcmp2-validate-response.png[WIS2 GDC online valida
...
+
+.Typical WCMP2 MQTT link
+[source,json]
+----
+{
+  "rel"  : "items",
+  "type" : "application/geo+json",
+  "title": "WIS2 notification service",
+  "href" : "mqtts://example.org",
+  "channel": "cache/a/wis2/ca-eccc-msc/data/core/weather/surface-based-observations"
```
Comment (amilan17) [inline cookbook/sections/data-publishers.adoc:143]:
```suggestion
  "channel": "cache/a/wis2/ca-eccc-msc/data/core/weather/surface-based-observations/synop"
```

Comment (amilan17) [reply cookbook/sections/data-publishers.adoc:143]: I recommend going to down to the last level possible.

File: cookbook/sections/data-publishers.adoc [hunk 2]
```diff
@@ -118,5 +118,187 @@ image::images/data-publishers-wcmp2-validate-response.png[WIS2 GDC online valida
...
+
+Given WCMP2, WTH and WNM, a user can subscribe to topics related to data of interest for download and access.
+
+In some cases, a dataset may be organized in a manner which requires additional further "filtering" such that a
+data consumer is only interested in a certain subset of the data granules being advertised by a given WNM.  Some examples include (but are not limited to), where a data consumer may be only be interested in:
+
+- surface weather observations from a certain station, or
+- numerical weather prediction forecast data for a certain timestep or weather parameter
+
+To implement this behaviour, add additional additional properties to both WCMP2 and WNM as follows:
```
Comment (amilan17) [inline cookbook/sections/data-publishers.adoc:155]:
```suggestion
To implement this behaviour, add additional properties to both WCMP2 and WNM as follows:
```

Comment (amilan17) [reply cookbook/sections/data-publishers.adoc:155]: or change to "more additional properties"?

**Review by amilan17 — COMMENTED** [review]
a couple suggestions inline

**Comment by tomkralidis** [comment]
Thanks for the feedback @antje-s @amilan17.  PR updated.

**Pull request merged**